service/matching/matching_engine.go

```diff
@@ -1222,19 +1196,18 @@ func (e *matchingEngineImpl) ForceUnloadTaskQueue(
 	ctx context.Context,
 	req *matchingservice.ForceUnloadTaskQueueRequest,
 ) (*matchingservice.ForceUnloadTaskQueueResponse, error) {
-	namespaceID := namespace.ID(req.GetNamespaceId())
-	taskQueue, err := newTaskQueueID(namespaceID, req.TaskQueue, req.TaskQueueType)
+	taskQueueFamily, err := tqid.NewTaskQueueFamily(req.NamespaceId, req.GetTaskQueue())
 	if err != nil {
 		return nil, err
 	}
-	tqm, err := e.getTaskQueuePartitionManager(ctx, taskQueue, normalStickyInfo, false)
+	pm, err := e.getTaskQueuePartitionManager(ctx, taskQueueFamily.TaskQueue(req.TaskQueueType).RootPartition(), true)
```

Comment:
this one seems wrong, though.. why are we assuming the root here? shouldn't this work on any partition?

Reply:
good catch. fixed.